### PR TITLE
Enable SAVE_ON_ALL_ACTIONS flag in browser tests

### PR DESCRIPTION
**Pull request by caitlinshk**
### Description

Enables the `SAVE_ON_ALL_ACTIONS` flag in browser tests and makes some slight modifications based on the failures:
 - Updates the settings screenshots to have it as enabled
 - Some places are now triggering the "error saving answers" modal when clicking "Review" or "Previous" on a block with unanswered required questions. Added some code to click "Continue without saving" on that modal so the test can proceed.
 -  Slight change to navigation around the "you're ineligible" page (explained more in a comment)
 - Update how the "Review" button is found (explained more in a comment)

Note that there are still many tests that manually disable the flag and test the old behavior: https://github.com/search?q=repo%3Aciviform%2Fciviform%20%22disableFeatureFlag(page%2C+%27save_on_all%22&type=code.

## Release notes

Note: This does **not** enable the SAVE_ON_ALL_ACTIONS feature for any staging or production environment, only the browser tests.

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)
- [x] Added an additional reviewer from outside your organization as FYI (if the primary reviewer is in the same organization as you)

#### New Features

- [x] Wrote browser tests with the feature flag off and on, etc.

### Issue(s) this completes

Part of #6450 
Feature flag issue is #6860

### CODE REVIEW

**Comment by caitlinshk** [comment]
* **#6868** <a href="https://app.graphite.dev/github/pr/civiform/civiform/6868?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @caitlinshk and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: browser-test/src/admin/admin_predicates.test.ts [hunk 1]
```diff
@@ -1214,18 +1216,22 @@ test.describe('create and edit predicates', () => {
       )
       // Validate that ineligible page is accessible.
       await validateAccessibility(page)
-      await page.goBack()
 
-      await applicantQuestions.clickReview()
```
Comment (caitlinshk) [inline browser-test/src/admin/admin_predicates.test.ts:1219]: Now that the "Review" button saves answers, if you click "Review" on a page that has ineligible answers then you'll be taken to the final "Based on your responses, you may not qualify" page. However, this test wants to test that the review page has the "you may not qualify" toast.

The "Go back and edit" button on the "Based on your responses, you may not qualify" page will always take you to the review page, so I updated this test to click that button so we can still get a screenshot of the review page with the "you may not qualify" toast.

File: browser-test/src/applicant/questions/phone.test.ts [hunk 2]
```diff
@@ -155,8 +155,8 @@ test.describe('phone question for applicant flow', () => {
 
       // Go back to question and ensure that "Review" button is also clickable
       // via Enter.
-      await page.click('a:has-text("Edit")')
-      await page.focus('a:has-text("Review")')
```
Comment (caitlinshk) [inline browser-test/src/applicant/questions/phone.test.ts:159]: When `SAVE_ON_ALL_ACTIONS` flag is enabled, the "Review" button is a `<button>` not a `<a>`, which is why this changed.

File: browser-test/src/admin/admin_predicates.test.ts (hunk 1 above)
Comment (swatkat1) [reply browser-test/src/admin/admin_predicates.test.ts:1219]: Thanks for the explaination!

**Review by swatkat1 — APPROVED** [review]
Thanks for the update

**Pull request merged**